pkg/controlplane/control/manager.go

```diff
@@ -81,7 +87,37 @@ type Manager struct {
 	lock            sync.Mutex
 	serviceToImport map[string]types.NamespacedName
 
-	logger *logrus.Entry
+	podLock sync.RWMutex
+	// endpoints stores the IPv4/v6 address of cl-dataplane endpoints per namespace
+	endpoints map[string][]discv1.Endpoint
+	logger    *logrus.Entry
+}
+
+func (m *Manager) addImportEndpointSlice(ctx context.Context, imp *v1alpha1.Import) error {
+	m.logger.Infof("Adding import endpointslice '%s/%s'.", imp.Namespace, imp.Name)
+
+	protocol := v1.ProtocolTCP
+	port32 := int32(imp.Spec.TargetPort)
+
+	es := discv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imp.Name + "-" + uuid.NewString()[:5],
```

Comment:
You should truncate imp.Name so that the entire string will not be more than 63 chars.

Reply:
Can you point me to the reference where its done for other imports, so that its consistent

Reply:
My bad, I see that endpoint slice names can be up to 253 chars.
Import names are actually limited to 63 (we should add a kubebuilder annotation for it).